### DISCUSSION TITLE
Compare/Contrast with OpenVSCode-Server

### DISCUSSION THREAD

**Discussion by colemickens** [eclipse-theia/theia#13899] [General] [answered]
One of the things that is most baffling to me is when folks build something similar to another tool and take no time to compare and contrast.

This seems to be true with Theia and OpenVSCode-Server. They seem *very* similar to me, and I struggle to understand why an end-user (not a white-labeler) would use Theia IDE over OpenVSCode-Server.

This is further exacerbated by the fact that Theia has, to my knowledge, meant at least two fairly distinct projects over the past ~5-6 years.

If I am a happy OpenVSCode-Server user, are there compelling reasons to switch to Theia IDE?

**Comment by colemickens** [comment]
Oh I wanted to note, when I try to Google this, Google offers me like 5 other searches that "other users also searched for". It's clear there are a LOT of people asking this question.

**Comment by JonasHelming** [comment, marked as answer]
That's a great question, and I'm currently working on an article that addresses this very topic. From an end-user perspective, both Theia and OpenVSCode-Server indeed share many similarities. However, there are distinct features and philosophies behind each that may influence your choice.
### Key Differences:
1. **Unique Features**:
   - **Theia**: Offers features not available in VS Code, such as a customizable toolbar and detachable views, which can enhance user experience.
   - **OpenVSCode-Server**: Provides the latest VS Code features and extensions, ensuring you have access to the most up-to-date tools.
2. **Privacy and Telemetry**:
   - The extent of privacy and telemetry in open forks of VS Code is a topic of ongoing debate. Users concerned with these aspects might prefer Theia for its community-driven approach that emphasizes openness and transparency.
3. **Extension Support**:
   - **OpenVSCode-Server**: Some VS Code extensions, such as remote container support and LiveShare, are proprietary and restricted to be used with VS Code only, and thus can't be used with products derived from the VS Code OSS codebase.
   - **Theia**: Provides open alternatives for these features, ensuring a completely open ecosystem.
4. **Forking Strategy and Control**:
   - **OpenVSCode-Server**: Aims to replicate VS Code's core features, which ties its future closely to the direction of the core VS Code project.
   - **Theia**: More community-driven with no single vendor controlling its development. This can lead to more diverse input and potentially faster adaptation to user needs.
### Considerations for Switching:
- **Satisfaction with Current Setup**:
  If you are satisfied with OpenVSCode-Server, there may not be a compelling reason to switch. It provides a familiar environment that closely mirrors VS Code.
- **Openness and Community Influence**:
  If you are looking for strategic independence from one single company, if you value open-source principles, want to avoid potential privacy issues, and wish to contribute to a vendor-neutral, community-driven project, Theia might be the better choice.
- **Feature Requirements**:
  Assess whether the unique features of Theia, such as its customizable interface and growing list of open solutions, align with your needs.


In summary:
- Stick with OpenVSCode-Server if you need a straightforward, reliable replication of VS Code and are content with its current offerings.
- Consider Theia if you prioritize a more open, community-led development model, strategic independence and are interested in features like customizable toolbars and detachable views.

For someone, who isn't an end-user but aims to provide an own product for their end-users, there are additional considerations, such as the additional flexibility for customizing the workbench, also see [Eclipse Theia vs. VS Code OSS](https://eclipsesource.com/blogs/2023/09/08/eclipse-theia-vs-code-oss/)
The landscape of these tools is evolving, and new developments could influence this comparison in the future. I hope this helps clarify the differences and assists in making an informed decision

**Reply by colemickens** [reply]
Nice, this confirms some suspicions. Very excited to see where Theia goes, and I'll likely take it for a spin again soon. Detachable views, replacements for the propriety bits (Remote, etc), and open governance are all pretty compelling to me.

Now to see if someone wants to write a NixOS module for it...

Thanks for your time and that detailed answer!

**Comment by JonasHelming** [comment]
Here is another article on [VS Code vs. the Theia IDE](https://eclipsesource.com/blogs/2024/07/12/vs-code-vs-theia-ide/)
